onnxruntime/core/framework/allocation_planner.h

```diff
@@ -71,9 +72,38 @@ class SequentialPlannerContext : public ISequentialPlannerContext {
   bool enable_memory_reuse_ = true;
 };
 
+// Given a graph with node placement information, partition the nodes into multiple sequence.
+// Each sequence can be executed in-dependently. The nodes in each sequence are executed in order,
+// but we can't assume any execution order between sequences, unless there is a data dependency.
+class IGraphPartitioner {
```

Comment:
I think the name is suggested by @pranavsharma . 
ISequencePartitioner seems good to me. @pranavsharma ?

Reply:
@skottmckay @souptc  Since we're still operating on the Graph, IGraphExecutionSequencer sounds good or IGraphExecutionPlanner as sequences is an implementation detail level concept of the planner.